packages/jupyter-ai/src/components/chat.tsx

```diff
@@ -35,6 +35,26 @@ type ChatBodyProps = {
   focusInputSignal: ISignal<unknown, void>;
 };
 
+/**
+ * Determines the name of the current persona based on the message history.
+ * Defaults to `'Jupyternaut'` if history is insufficient.
+ */
+function getPersonaName(messages: AiService.ChatMessage[]): string {
+  let personaName = 'Jupyternaut';
+
+  for (let i = messages.length - 1; i >= 0; i--) {
+    const message = messages[i];
+    if (message.type !== 'agent' && message.type !== 'agent-stream') {
+      continue;
+    }
+
+    personaName = message.persona.name;
+    break;
```

Comment:
To simplify control flow and improve readability, use `(message.type == 'agent' || message.type == 'agent-stream')` for condition and immediately return `message.persona.name` if it is true.

Reply:
Thanks for the suggestion! I've implemented it.